### PR TITLE
Add ability to specify custom flags

### LINKED ISSUE #1: Add ability to specify custom flags for upx and strip
Currently we're just using the default which could be improved.

### PR DESCRIPTION
**Pull request by crazy-max**
Fixes #1 and also use `@actions/exec` instead of `child_process` dep

### CODE REVIEW

**Comment by svenstaro** [comment]
Awesome! I like the usage of `@actions/exec`! Could you also add the for the `strip` part in order to solve #1 in its entirety?

**Comment by svenstaro** [comment]
Perhaps you could also add a new complex example using these args to the end of the README?

**Comment by crazy-max** [comment]
> Could you also add the for the strip part in order to solve #1 in its entirety?

I wonder what's the purpose of `strip` here ? UPX already handles this and this adds to this action another layer that could be unexpected for the user.

> Perhaps you could also add a new complex example using these args to the end of the README?

Yes of course

**Comment by svenstaro** [comment]
In my tests, no `strip`ping is done by UPX and the binary ends up a lot bigger without stripping.

**Comment by crazy-max** [comment]
@svenstaro I have just checked, you right! So I think we could add an input to enable/disable `strip` command because for Windows binaries `strip` could lead to unexpected behavior.

**Comment by svenstaro** [comment]
So far in my tests, strip was always available in GitHub Actions and is working as intendend.

**Comment by crazy-max** [comment]
I talked about final result when a Windows executable is stripped.
For example if I execute `strip` on `innosetup-5.6.1-unicode.exe` the stripped executable is broken.

**Comment by svenstaro** [comment]
I use this action [here](https://github.com/svenstaro/dummyhttp/releases) and it seems fine.

**Comment by crazy-max** [comment]
@svenstaro Ok I have made some changes ;)

File: README.md [hunk 1]
```diff
@@ -5,9 +5,10 @@ It runs on all operating systems types offered by GitHub.
 
 ## Input variables
 
-You must provide:
-
-- `file`: The file you want to compress. It's compressed in-place.
+* `file`: The file you want to compress. It's compressed in-place. **required**
+* `args`: Arguments to pass to UPX. *optional*
+* `strip`: Whether or not "strip" symbols from object file. *optional*
```
Comment (svenstaro) [inline README.md]: You should document the default being `true`.

Comment (crazy-max) [reply README.md]: Done

**Comment by svenstaro** [comment]
Good changes so far, but I'm still missing the second example for the README with the args in use.

File: lib/action.js [hunk 2]
```diff
@@ -46,21 +46,24 @@ function downloadUpx() {
...
         try {
             const file = core.getInput('file', { required: true });
+            const args = core.getInput('args');
+            const strip = core.getInput('strip') || 'true';
+            const strip_args = core.getInput('strip_args');
             if (!fs.existsSync(file)) {
-                core.setFailed(`File ${file} wasn't found.`);
+                core.setFailed(`⛔ File ${file} wasn't found.`);
+            }
+            if (/true/i.test(strip)) {
```
Comment (svenstaro) [inline lib/action.js]: I've never seen this syntax, what is this?

File: src/action.ts [hunk 3]
```diff
@@ -26,21 +26,27 @@ async function downloadUpx(): Promise<string> {
...
+        const args = core.getInput('args');
+        const strip = core.getInput('strip') || 'true';
+        const strip_args = core.getInput('strip_args');
 
         if (!fs.existsSync(file)) {
-            core.setFailed(`File ${file} wasn't found.`);
+            core.setFailed(`⛔ File ${file} wasn't found.`);
         }
 
+        if (/true/i.test(strip)) {
```
Comment (svenstaro) [inline src/action.ts:40]: I've never seen this syntax. What is this?

File: src/action.ts [hunk 4]
```diff
@@ -26,21 +26,27 @@ async function downloadUpx(): Promise<string> {
...
+            console.log('🏃 Running strip...');
```
Comment (svenstaro) [inline src/action.ts]: I think you're supposed to use [info()](https://github.com/actions/toolkit/blob/master/packages/core/src/core.ts#L138) instead but I'm not sure. Opinions?

File: src/action.ts (hunk 3 above)
Comment (crazy-max) [reply src/action.ts:40]: It's the safest way to check if a string is a true boolean

File: src/action.ts (hunk 4 above)
Comment (crazy-max) [reply src/action.ts]: Yeah maybe we should!

**Comment by svenstaro** [comment]
Seems good to me. Once the README is expanded this seems good to merge.

**Comment by crazy-max** [comment]
@svenstaro I have updated README with a complete example of the available inputs.

**Comment by svenstaro** [comment]
I'd prefer a second example. One for the simple case and one for the complex case. For reference, see https://github.com/svenstaro/upload-release-action

**Comment by crazy-max** [comment]
Done 🙂

**Comment by svenstaro** [comment]
Great stuff!

**Comment by svenstaro** [comment]
Released as 1.1.0.

**Comment by crazy-max** [comment]
Yeah :rocket: :tada:

**Pull request merged**